src/redux/missions/missions.js

```diff
@@ -17,7 +25,16 @@ export const getMissionFromApi = () => async (dispatch) => {
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case FETCH_MISSIONS:
-      return [...action.payload];
+      return { ...state, missions: action.payload };
+    case CREATE_JOIN: {
+      const changeJoin = state.missions.map((el) => {
+        if (el.mission_id === action.payload.mission_id) {
+          return { ...el, join: action.payload.join };
+        }
+        return el;
+      });
+      return { ...state, missions: changeJoin };
+    }
     default:
       return state;
```

Comment:
bro your code is great but it would be better if you  fix  the navigation preserve issues, the cause of this issue is that react rerender the component whenever you
navigate you can solve it using flag
![image](https://user-images.githubusercontent.com/74744787/151156732-665a3707-fe04-48b4-95b7-4ff0baae626f.png)
you can check my solution to that problem:
![image](https://user-images.githubusercontent.com/74744787/151156938-eadcb8d8-31a0-485a-8392-ae879d261bff.png)